internal/services/cdn/cdn_endpoint_custom_domain_resource_test.go

```diff
@@ -139,6 +240,100 @@ resource "azurerm_cdn_endpoint_custom_domain" "import" {
 `, template)
 }
 
+func (r CdnEndpointCustomDomainResource) httpsCdn(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+%s
+resource "azurerm_cdn_endpoint_custom_domain" "test" {
+  name            = "acctest-customdomain"
+  cdn_endpoint_id = azurerm_cdn_endpoint.test.id
+  host_name       = "${azurerm_dns_cname_record.test.name}.${data.azurerm_dns_zone.test.name}"
+  cdn_managed_https {
+    certificate_type = "Dedicated"
+    protocol_type    = "IPBased"
+  }
+}
+`, template)
+}
+
+func (r CdnEndpointCustomDomainResource) httpsUserManaged(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+%[1]s
+data "azurerm_client_config" "test" {
+}
+resource "azurerm_key_vault" "test" {
+  name                = "testkeyvault-%[2]d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  tenant_id           = data.azurerm_client_config.test.tenant_id
+  sku_name            = "standard"
+  access_policy {
+    tenant_id = data.azurerm_client_config.test.tenant_id
+    object_id = data.azurerm_client_config.test.object_id
+    certificate_permissions = [
+      "get",
+      "delete",
+      "import",
+      "purge",
+    ]
+    key_permissions = [
+      "get",
+      "create",
+    ]
+    secret_permissions = [
+      "get",
+      "set",
+    ]
+  }
+  access_policy {
+    tenant_id = data.azurerm_client_config.test.tenant_id
+    object_id = "4dbab725-22a4-44d5-ad44-c267ca38a954" # The Microsoft Azure.Cdn application object ID
```

Comment:
Presumably this is different in regions other than Azure Public? if so, can we look this up?

Reply:
Indeed, this is even different between different tenant :sweat_smile: